src/Core/Module/ModuleRepository.php

```diff
@@ -156,7 +159,9 @@ public function getModule(string $moduleName): ModuleInterface
 
         $this->cacheProvider->save($cacheKey, new Module($attributes, $disk, $database));
 
-        return $this->cacheProvider->fetch($cacheKey);
+        $coreModule = $this->cacheProvider->fetch($cacheKey);
+
+        return $this->enrichModuleAttributesFromHook($coreModule);
```

Comment:
I'm not sure it's relevant to get back the module via the cache since you created it the line before Instead you can simply store it in a variable
```suggestion
        $coreModule = new Module($attributes, $disk, $database);
        $this->cacheProvider->save($cacheKey, $coreModule);

        return $this->enrichModuleAttributesFromHook($coreModule);
```

Reply:
:thinking:  As it was already done this way, can't it be a way to validate that the Module is well stored in cache ?

For example, is it possible that `$this->cacheProvider->save` fails ? 

:thinking:  But I suppose if we use this "trick" to detect a cache problem, it should be considered like a cache issue that we should fix independently ...

Whatever ! I'll apply the change as I consider your code better :)